precompiles/pallet-xcm/src/lib.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright Moonsong Labs
+// This file is part of Moonkit.
+
+// Moonkit is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonkit is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonkit.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use fp_evm::PrecompileHandle;
+use frame_support::{
+	dispatch::{GetDispatchInfo, PostDispatchInfo},
+	traits::ConstU32,
+};
+use pallet_evm::AddressMapping;
+use precompile_utils::prelude::*;
+
+use sp_core::U256;
+use sp_runtime::traits::Dispatchable;
+use sp_std::marker::PhantomData;
+use sp_weights::Weight;
+use xcm::{
+	latest::{Asset, AssetId, Assets, Fungibility, Location},
+	prelude::WeightLimit::*,
+	VersionedAssets, VersionedLocation,
+};
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub const MAX_ASSETS_ARRAY_LIMIT: u32 = 2;
```

Comment:
Isnt this retrievable through the config?¿

Reply:
Nop, there is the const `MAX_ASSETS_FOR_TRANSFER` inside pallet-xcm but it's not public: https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/pallet-xcm/src/lib.rs#L1450.

Reply:
I don't think so. There is the const `MAX_ASSETS_FOR_TRANSFER` inside pallet-xcm but it's not public: https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/pallet-xcm/src/lib.rs#L1450.